machineconfiguration/v1/types_machineconfignode.go

```diff
@@ -209,6 +224,39 @@ type MachineConfigNodeSpecMachineConfigVersion struct {
 	Desired string `json:"desired"`
 }
 
+// MachineConfigNodeSpecConfigImage holds the desired image for the node.
+// This structure is populated from the `machineconfiguration.openshift.io/desiredImage`
+// annotation on the target node, which is set by the Machine Config Pool controller
+// to signal the desired image pullspec for the node to update to.
+type MachineConfigNodeSpecConfigImage struct {
+	// desiredImage is the fully-qualified pullspec of the image that the Machine
+	// Config Operator (MCO) intends to apply to the node.
+	// +kubebuilder:validation:MaxLength:=253
+	// +required
```

Comment:
explicitly include the constraints in the GoDoc - users can't see the markers here in generated documentation so if we do not explicitly mention that the field is required and cannot have a length greater than 253 characters users won't know until they attempt to create a resource that fails.

Additionally, what is a fully-qualified pullspec? are there constraints we can enforce _beyond_ the current ones that ensure malformed pullspecs are not being provided by a user? Having more concrete validation in place here helps have a faster feedback loop for end-users as a malformed pullspec would be rejected at admission time rather than resulting in an operator side status condition being set.